netcore/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+project(netcore LANGUAGES NONE)
+
+option(INCLUDE_DOTNET_CORE "Build .NET Core client." ON)
+
+find_program(DOTNET dotnet)
+
+if(DOTNET AND INCLUDE_DOTNET_CORE)
+  add_custom_target(netcore ALL COMMAND ${DOTNET} build --configuration $<CONFIG> WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/NetCore DEPENDS apache-geode-c VERBATIM)
+  add_custom_target(netcore-test ALL COMMAND ${DOTNET} build --configuration $<CONFIG> WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/NetCore.Test DEPENDS netcore VERBATIM)
+
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/NetCore/bin/$<CONFIG>/netcoreapp3.1/Apache.Geode.NetCore.dll DESTINATION ./lib)
```

Comment:
I believe this is correct, yes.  Symbols also live in .pdb files on all platforms, IIRC.  I don't recall how library resolution works in .net core on non-Windows platforms, tho.

Reply:
Should we be copying the PDB files then too?

Reply:
> Do all .NET core assemblies have .dll suffixes on all platforms?

Unfortunately yes, they do.

> In Windows it was the norm for .dll files to go in the bin directory since they have to be in the runtime path. Is that not the case anymore? What about on other platforms, do they search the runtime path or the library path?

We install to *a* directory. Typically the user can specify a bin directory in their path if they want, if they're building from source. We build the pivnet zip/tar file from the contents of the install directory in the CI.